sdk/identity/azure-identity/src/client_certificate_credential.cpp

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/identity/client_certificate_credential.hpp"
+
+#include "private/token_credential_impl.hpp"
+
+#include <azure/core/base64.hpp>
+#include <azure/core/datetime.hpp>
+#include <azure/core/uuid.hpp>
+
+#include <chrono>
+#include <iomanip>
+#include <sstream>
+#include <utility>
+#include <vector>
+
+#include <openssl/bio.h>
+#include <openssl/evp.h>
+#include <openssl/ossl_typ.h>
+#include <openssl/pem.h>
+#include <openssl/rsa.h>
+#include <openssl/x509.h>
+
+using namespace Azure::Identity;
+
+namespace {
+template <typename T> std::vector<uint8_t> ToUInt8Vector(T const& in)
+{
+  const size_t size = in.size();
+  std::vector<uint8_t> outVec(size);
+  for (size_t i = 0; i < size; ++i)
+  {
+    outVec[i] = static_cast<uint8_t>(in[i]);
+  }
```

Comment:
nit. Wondering if you can use the vector's constructor:
```
template< class InputIt >
vector( InputIt first, InputIt last,
        const Allocator& alloc = Allocator() );
```

You wouldn't need the for loop in that case, like:

```cpp
std::vector<uint8_T> outVec(in.begin(), in.end());
```

You wouldn't need the size anymore.

Reply:
Unfortunately, it will generate warnings, because one input vector is of `unsigned char` and in another case it is of `char`, but we are building a vector of `uint8_t`. That constructor _can_ be used, but most probably with warning suppression, or with reinterpret-casting.